### PR TITLE
Stream http requests to the data endpoint when downloading Json files.

### PR DESCRIPTION
**Pull request by JohnMwashuma**

### CODE REVIEW

**Comment by JohnMwashuma** [comment]
@pld Am seeing this notification when a login Codacy on why Codacy is unable to analyse my pr.

screen shot 2018-08-28 at 10 23 55

File: src/milia/api/io.clj [hunk 1]
```diff
@@ -86,21 +86,36 @@
...
   "Parse binary response by writing into a temp file and returning the path."
-  [body filename]
+  [body filename & {:keys [url http-options]}]
   (let [tempfile (File/createTempFile filename "")
         path (str (.getAbsolutePath tempfile))
-        ^File file (clojure.java.io/file path)]
+        ^File file (clojure.java.io/file path)
+        ;; Stream the http-request to avoid out of memory errors when the data
+        ;; to copy is large
+        streamed-http-request
```
Comment (ivermac) [inline src/milia/api/io.clj]: We could destructure `streamed-http-request` here and use `status` and `body` in lines `101`, `102` and `107`.

File: src/milia/utils/file.clj [hunk 2]
```diff
@@ -2,6 +2,8 @@
   (:require [clojure.java.io :as io])
   (:import [org.apache.commons.io IOUtils]))
 
+(def file-extension-regex #"\w+$")
```
Comment (ivermac) [inline src/milia/utils/file.clj]: This regex pattern seems to match any word. Which file extension were we targeting?

Comment (JohnMwashuma) [reply src/milia/utils/file.clj]: We are just targeting json file extensions.

Comment (JohnMwashuma) [reply src/milia/utils/file.clj]: Let me check if i can one that is specific to this.

Comment (JohnMwashuma) [reply src/milia/utils/file.clj]: Will use this instead `#".json$"`

File: src/milia/api/io.clj [hunk 3]
```diff
@@ -86,21 +86,36 @@
...
-          (.write w body-bytes))))
+    ;; io/copy is used since it takes an input-stream and an output-stream
+    (if (and json-file? (not (error-status? status)))
+      (with-open [in-stream streamed-body
+                  out-stream (->> file
+                                  io/as-file
+                                  io/output-stream)]
+        (io/copy in-stream out-stream))
+      (parse-json-response streamed-body))
+      ;; Broken out so we can add type hints to avoid reflection
```
Comment (pld) [inline src/milia/api/io.clj]: out-dent to align

**Pull request merged**